### ISSUE TITLE
Current Status

### ISSUE THREAD

**Issue by PashPaw** [PDP-10/joss-36#33]
So, I got it to compile.  Sort of.  

It's not willing to run, however.  There are still too many missing symbols for it to be functional.  We might want to get a hold of RAND to see if they have anything to help us like more listings.

If not, it's going to take some serious reverse engineering to get it to work.  It looks like KMON covers both the supervisor and distributor.

**Comment by larsbrinkhoff** [comment]
I'm working on RAND, but so far I haven't got anything beyond those two
additional documents.

**Comment by PashPaw** [comment]
Okay.  Thanks.  The CHM has some documents of historical significance (as you likely know) but trying to get anything out of them is going to be impossible.

Things born digital should be made easily digital.

**Comment by larsbrinkhoff** [comment]
Do you have a list?  I may have a favor to call in at the CHM.

**Comment by PashPaw** [comment]
https://www.computerhistory.org/collections/catalog/102734547
https://www.computerhistory.org/collections/catalog/102679426

>

**Comment by larsbrinkhoff** [comment]
One thing that remains to be done, is to compare the addresses in the new listing with the old.  I don't think they match yet.

**Comment by PashPaw** [comment]
True.  Do you think the extra comments could be causing that or something else?

**Comment by PashPaw** [comment]
This is peculiar.  Pg. 41 doesn't seem to be missing any instructions and most of the page matches except for the last few instructions.

**Comment by dabridgham** [comment]
I don't have the combined file you're assembling but it looks like in the file I typed in I missed one instruction.  There's supposed to be a "jsr cvtl1" just before the "aos tint(c)".

**Comment by PashPaw** [comment]
Thanks.  Go ahead and correct it.

I've just been just joining different files together with cat and using a tool included with simh called asc to process the line endings so TOPS-10 can read them properly after using the simulated KA10's paper tape reader (it's a stupid trick that also works with the PDP-11 sims).

I don't know what @larsbrinkhoff is doing to get them in.  He's probably using a file he's created on his own and we're coming up with the same results.

Unprocessed (jos_su.mac): [jos_su.pt.txt](https://github.com/PDP-10/joss-36/files/4302760/jos_su.pt.txt)
.
Processed (jos_su.pt): [jos_su.txt](https://github.com/PDP-10/joss-36/files/4302758/jos_su.txt)

**Comment by PashPaw** [comment]
Never mind.  Added.

**Comment by PashPaw** [comment]
On TOPS-10, using + instead of commas seems to produce fewer errors.  I might need to see why this is.

I might have to see what the <> notation does.

**Comment by larsbrinkhoff** [comment]
All I did was `cat page*.mac > joss.mac` and then edited joss.mac to have CRLF line endings.

Maybe it's time to go from the separate pageNNN files to a single su.mac?

**Comment by PashPaw** [comment]
Maybe but the separate files are easier to edit.

Here’s a compromise: produce a single file that will be updated when the pages get fixed.

**Comment by larsbrinkhoff** [comment]
My CHM agent says he's working on it.  Especially the first item looks interesting because the description includes "program listing".

**Comment by PashPaw** [comment]
Cool!  Thanks.

So, my hunch was likely correct?

**Comment by larsbrinkhoff** [comment]
They say it's 30 cents per scanned page.

**Comment by PashPaw** [comment]
And how many pages?

**Comment by larsbrinkhoff** [comment]
I don't know, that's all I heard so far.

**Comment by larsbrinkhoff** [comment]
CHM quoted $6000 for scanning their documents.

**Comment by PashPaw** [comment]
They’re out of their mind.  That reads like they don’t want to do it.

**Comment by larsbrinkhoff** [comment]
I contacted RAND Archives again through their [online form](https://www.rand.org/about/history/archives/archives-contact-form.html) and got this back:
> Thank you for contacting the RAND Corporation Archives again. Since you're interested in taking at look at more public material on RAND's JOSS program, I've attached a custom bibliography of all of RAND's public papers and reports on JOSS to this email.
> 
> For all of the papers and reports not currently available online for free on our www.rand.org website, you can order them through our Customer Services department at order@rand.org. If you're only looking for a few papers that aren't online yet, I can scan them for you as a courtesy. Just let me know which ones you're most interested in.

[JOSS Bibliography.docx](https://github.com/PDP-10/joss-36/files/7103616/JOSS.Bibliography.docx)

**Comment by larsbrinkhoff** [comment]
Going through that bibliography, I see some things from around 1967 that seem of interest:

- JOSS: User Scheduling and Resource Allocation. Santa Monica, CA: RAND Corporation, RM-5216-PR
- JOSS: Accounting and Performance Measurement. Santa Monica, CA: RAND Corporation, RM-5217-PR
- JOSS: Introduction to the System Implementation. Santa Monica, CA: RAND Corporation, P-3486

Everything else is either already online, or is not related to the PDP-6 implementation.

**Comment by PashPaw** [comment]
> Going through that bibliography, I see some things from around 1967 that seem of interest:
> 
> * JOSS: User Scheduling and Resource Allocation. Santa Monica, CA: RAND Corporation, RM-5216-PR
> * JOSS: Accounting and Performance Measurement. Santa Monica, CA: RAND Corporation, RM-5217-PR
> * JOSS: Introduction to the System Implementation. Santa Monica, CA: RAND Corporation, P-3486
> 
> Everything else is either already online, or is not related to the PDP-6 implementation.

Correct.  In fact, those are the bits we likely wanted.

Also, the Disc File System and Distributor documents contain a weird mix of pseudocode and machine code in them. It'll be an undertaking but we might be able to resurrect them. 

There may be some other difficulties but we'll get to that when we get there.  In the meantime, it's probably time for me to start studying MACRO-10 again with whatever else I'm studying ATM.

**Comment by larsbrinkhoff** [comment]
So I'll ask for those three documents.  After that, I guess we'll have to roll our own or get code form AID for the components we don't have original JOSS-II source code for.

**Comment by larsbrinkhoff** [comment]
Hey all,

I received those three documents.  They are now on a branch called lars/rand.

**Comment by larsbrinkhoff** [comment]
Note, the documents are not available for free from the Rand website so please don't distribute.

**Comment by PashPaw** [comment]
Cool!  I’ll merge them later.

**Comment by PashPaw** [comment]
Or not.  But okay.

**Comment by PashPaw** [comment]
Oh.  And thanks to both you and RAND, Lars.  I don’t think I would have gotten as far without help.

So, from a cursory look in the JOSS: User Scheduling and Allocation, there is a single piece of non-DEC hardware mentioned because of the line concentrator.  It’s apparently custom and based around the Stowager switch.  It then inputs into a line the 630 if I’m reading things right.

**Comment by PashPaw** [comment]
My current assumption is that kmon was a shim for the monitor.  I could be wrong but the AID code does mention some weird things.  I should figure out how to extract the code from the TOPS-10 7.03/7.04 tapes and upload it since segments were removed.

There's a mention of DDT in it, which, even for Monitor 5.03, would have been odd.    AID returned to the Monitor level and never to DDT on Monitor/TOPS-10 and TOPS-20.  I'm also tempted to rename the repo to JOSS-II, ATM.
